ts/components/ui/activityIndicator.tsx

```diff
@@ -0,0 +1,19 @@
+import * as React from "react";
+import { ActivityIndicator as RNActivityIndicator } from "react-native";
+
+import variables from "../../theme/variables";
+
+interface Props {
+  color?: string;
+}
+
+const ActivityIndicator: React.SFC<Props> = ({ color }) => (
+  <RNActivityIndicator size="large" color={color} />
```

Comment:
to avoid the object mutation you could just fallback to the default value here: `color={color || variables.brandPrimary}`

Reply:
Why not use `defaultProps`? It's a React functionality, I wouldn't change that just to avoid a linter warning/error.

Reply:
That's true, but one of the principles guiding this project is to favor immutability and a functional approach - in this case, given that we have the option, the aforementioned principle would dictate to avoid the defaultProps in favor of the default value in the parameter